### PR TITLE
Fix spelling in docblocks and removed @return labels in constructors.

### PR DESCRIPTION
**Pull request by mbabker**
Using the code inspector in PhpStorm, evaluated the following conditions from the results:
- Spelling errors (specifically in code comments and doc blocks)
- A method declared void should not return anything
- Doc block doesn't match method declaration

There was also an instance of DatabaseException that I saw while looking through the results that I switched to JDatabaseException.

### CODE REVIEW

**Comment by elkuku** [comment]
Quite impressive :)
What I love most are the changes of `@return void` to the proper classes - two thumbs up ;)

File: libraries/joomla/database/database.php [hunk 1]
```diff
@@ -432,7 +432,7 @@ abstract class JDatabase implements JDatabaseInterface
 	 *
 	 * @param   array  $options  List of options used to configure the connection
 	 *
-	 * @return  void
+	 * @return  JDatabase
```
Comment (eddieajau) [inline libraries/joomla/database/database.php]: Based on previous cases, this should be void (it's the __construct method).

File: libraries/joomla/database/database/mysql.php [hunk 2]
```diff
@@ -56,7 +56,7 @@ class JDatabaseMySQL extends JDatabase
 	 *
 	 * @param   array  $options  Array of database options with keys: host, user, password, database, select.
 	 *
-	 * @return  void
+	 * @return  JDatabaseMySQL
```
Comment (eddieajau) [inline libraries/joomla/database/database/mysql.php]: Should be return void

File: libraries/joomla/database/database/mysqli.php [hunk 3]
```diff
@@ -56,7 +56,7 @@ class JDatabaseMySQLi extends JDatabase
 	 *
 	 * @param   array  $options  List of options used to configure the connection
 	 *
-	 * @return  void
+	 * @return  JDatabaseMySQLi
```
Comment (eddieajau) [inline libraries/joomla/database/database/mysqli.php]: Should be return void

File: libraries/joomla/database/database/sqlsrv.php [hunk 4]
```diff
@@ -69,7 +69,7 @@ class JDatabaseSQLSrv extends JDatabase
 	 *
 	 * @param   array  $options  List of options used to configure the connection
 	 *
-	 * @return  void
+	 * @return  JDatabaseSQLSrv
```
Comment (eddieajau) [inline libraries/joomla/database/database/sqlsrv.php]: Should be return void

File: libraries/joomla/environment/browser.php [hunk 5]
```diff
@@ -193,7 +193,7 @@ class JBrowser extends JObject
 	 * @param   string  $userAgent  The browser string to parse.
 	 * @param   string  $accept     The HTTP_ACCEPT settings to use.
 	 *
-	 * @return  void
+	 * @return  JBrowser
```
Comment (eddieajau) [inline libraries/joomla/environment/browser.php]: Should be return void

File: libraries/joomla/filesystem/archive/bzip2.php [hunk 6]
```diff
@@ -31,7 +31,7 @@ class JArchiveBzip2 extends JObject
 	/**
 	 * Constructor tries to load the bz2 extension if not loaded
 	 *
-	 * @return  void
+	 * @return  JArchiveBzip2
```
Comment (eddieajau) [inline libraries/joomla/filesystem/archive/bzip2.php]: Should be return void

File: libraries/joomla/form/form.php [hunk 7]
```diff
@@ -78,7 +78,7 @@ class JForm
 	 * @param   string  $name     The name of the form.
 	 * @param   array   $options  An array of form options.
 	 *
-	 * @return  void
+	 * @return  JForm
```
Comment (eddieajau) [inline libraries/joomla/form/form.php]: Should be return void

File: libraries/joomla/html/pagination.php [hunk 8]
```diff
@@ -66,7 +66,7 @@ class JPagination extends JObject
 	 * @param   integer  $limit       The number of items to display per page.
 	 * @param   string   $prefix      The prefix used for request variables.
 	 *
-	 * @return  void
+	 * @return  JPagination
```
Comment (eddieajau) [inline libraries/joomla/html/pagination.php]: Should be return void

File: libraries/joomla/html/pagination.php [hunk 9]
```diff
@@ -743,7 +743,7 @@ class JPaginationObject extends JObject
 	 * @param   integer  $base    The number of rows as a base offset.
 	 * @param   string   $link    The link URL.
 	 *
-	 * @return  void
+	 * @return  JPaginationObject
```
Comment (eddieajau) [inline libraries/joomla/html/pagination.php]: Should be return void

File: libraries/joomla/log/logentry.php [hunk 10]
```diff
@@ -77,7 +77,7 @@ class JLogEntry
 	 * @param   string  $category  Type of entry
 	 * @param   string  $date      Date of entry (defaults to now if not specified or blank)
 	 *
-	 * @return  void
+	 * @return  JLogEntry
```
Comment (eddieajau) [inline libraries/joomla/log/logentry.php]: Should be return void

File: libraries/joomla/log/logger.php [hunk 11]
```diff
@@ -35,7 +35,7 @@ abstract class JLogger
 	 *
 	 * @param   array  &$options  Log object options.
 	 *
-	 * @return  void
+	 * @return  JLogger
```
Comment (eddieajau) [inline libraries/joomla/log/logger.php]: Should be return void

File: libraries/joomla/log/loggers/database.php [hunk 12]
```diff
@@ -73,7 +73,7 @@ class JLoggerDatabase extends JLogger
 	 *
 	 * @param   array  &$options  Log object options.
 	 *
-	 * @return  void
+	 * @return  JLoggerDatabase
```
Comment (eddieajau) [inline libraries/joomla/log/loggers/database.php]: Should be return void

File: libraries/joomla/log/loggers/formattedtext.php [hunk 13]
```diff
@@ -69,7 +69,7 @@ class JLoggerFormattedText extends JLogger
 	 *
 	 * @param   array  &$options  Log object options.
 	 *
-	 * @return  void
+	 * @return  JLoggerFormattedText
```
Comment (eddieajau) [inline libraries/joomla/log/loggers/formattedtext.php]: Should be return void

File: libraries/joomla/log/loggers/syslog.php [hunk 14]
```diff
@@ -45,7 +45,7 @@ class JLoggerSysLog extends JLogger
 	 *
 	 * @param   array  &$options  Log object options.
 	 *
-	 * @return  void
+	 * @return  JLoggerSysLog
```
Comment (eddieajau) [inline libraries/joomla/log/loggers/syslog.php]: Should be return void

**Comment by mbabker** [comment]
So what's "correct"?  There's no consistency right now with the constructor `@return` statements.  In the couple dozen I've just looked at, there's a mix between void, the class name, and no return statement at all, and it seems to lean closer to being a majority class names.

File: libraries/joomla/log/loggers/w3c.php [hunk 15]
```diff
@@ -39,7 +39,7 @@ class JLoggerW3C extends JLoggerFormattedText
 	 *
 	 * @param   array  &$options  Log object options.
 	 *
-	 * @return  void
+	 * @return  JLoggerW3C
```
Comment (eddieajau) [inline libraries/joomla/log/loggers/w3c.php]: Should be return void

**Comment by eddieajau** [comment]
Yeah, that's mostly my fault.  The method strictly does not return anything so it should be void.

File: libraries/joomla/registry/registry.php [hunk 16]
```diff
@@ -34,7 +34,7 @@ class JRegistry
 	 *
 	 * @param   mixed  $data  The data to bind to the new JRegistry object.
 	 *
-	 * @return  void
+	 * @return  JRegistry
```
Comment (eddieajau) [inline libraries/joomla/registry/registry.php]: Should be return void

File: libraries/joomla/utilities/date.php [hunk 17]
```diff
@@ -82,7 +82,7 @@ class JDate extends DateTime
 	 * @param   string  $date  String in a format accepted by strtotime(), defaults to "now".
 	 * @param   mixed   $tz    Time zone to be used for the date.
 	 *
-	 * @return  void
+	 * @return  JDate
```
Comment (eddieajau) [inline libraries/joomla/utilities/date.php]: Should be return void

File: libraries/joomla/utilities/simplecrypt.php [hunk 18]
```diff
@@ -31,7 +31,7 @@ class JSimpleCrypt extends JObject
 	 *
 	 * @param   string  $key  Optional encryption key
 	 *
-	 * @return  void
+	 * @return  JSimpleCrypt
```
Comment (eddieajau) [inline libraries/joomla/utilities/simplecrypt.php]: Should be return void

**Comment by mbabker** [comment]
I think an argument can be made for making the `@return` tags consistent one way or another here.  Personally, I'm in favor of either using the class names or not including a `@return` tag on constructors.

Running phpDocumentor locally on my own component using `@return void`, `@return ClassName`, and no `@return` statement produces different results in the output (this is pretty much the default, so unless the API's implementation has been tweaked in such a way to do something different, it should apply).

`void __construct( [array $config = array()])` - constructor with `@return void`
`PodcastManagerTableFeed __construct( object &$db)` - constructor with `@return PodcastManagerTableFeed` and class name linked to self
`PodcastManagerModelFeeds __construct( [array $config = array()])` - constructor with no `@return` statement

Though the method does not explicitly return something, it does create an instance of a given class, and it seems `@return void` isn't the correct answer for the constructors.

**Comment by eddieajau** [comment]
Ok, scouting around I think you are right.  Seems the consensus is to **not** use `@return` in constructors - I'm fine with that.

**Comment by elinw** [comment]
This was something i was struggling with as I went through because there was so much inconsistency but I believe it is really not quite proper to have a constructor return something. You don't say $something = JSimpleCrypt so it's not exactly a return in the same way that a regular method is.

**Comment by mbabker** [comment]
I'll handle the `@return` statements tonight and tomorrow.  Question - Will this require a change to the code sniffer, and if so, do you think you could handle it @elkuku?

**Comment by elkuku** [comment]
I didn't even noticed that the `@return` statements belong to constructors - bad github diff... ;(
So.. I would assume that the same goes also for destructors ? (just in case)
That would be something like:
- **Standard**: Constructor and destructor comments must not have a `@return` tag.

If this is OK I can modify the corresponding sniff.

**Comment by mbabker** [comment]
Request updated removing the `@return` statements.

**Comment by eddieajau** [comment]
Awesome work Michael. Really appreciate it.

**Pull request merged**